libs/blocks/modal/modal.js

```diff
@@ -120,6 +127,16 @@ export async function getModal(details, custom) {
 
   dialog.append(close);
   document.body.append(dialog);
+  dialog.insertAdjacentElement('afterend', curtain);
+  document.getElementsByClassName("dialog-close")[0]?.focus({ focusVisible: true });
+
+  const closeModal = (event) => {
+    var keyPressed = event.key;
+    if (keyPressed?.toLowerCase() === 'escape') {
+      document.removeEventListener('keydown', closeModal);
+    }
+  }
+  document.addEventListener('keydown', closeModal);
```

Comment:
if you're going to add an event listener to the document, you should probably also remove the event listener on a modal close :)